File: xdm-app/src/main/java/xdm/app/service/AppConfigService.java
package xdm.app.service;

import xdm.app.constants.SortKey;
import xdm.core.downloaders.Metadata;

import java.util.List;

public interface AppConfigService {
  void load();

  boolean shouldShowDownloadCompleteWindow();

  boolean shouldRunVirusScan();

  boolean shouldRunCommand();

  boolean shouldShowDownloadProgressWindow();

  String getDefaultDownloadFolder();

  String getTempFolder();

  boolean shouldAutoRenameOnConflict();

  boolean shouldShutdownAfterAllDone();

  int getMaxParallelDownloads();

  List<String> getRecentFolders();

  boolean isAutoSelectFolder();

  void setAutoSelectFolder(boolean value);

  int getFolderIndex();

  void setFolderIndex(int value);

  SortKey getSortKey();

  void setSortKey(SortKey sortKey);

  boolean isSortAscending();

  void setSortAscending(boolean ascending);

  long getMinVideoSize();
  void setMinVideoSize(long value);
}

File: xdm-app/src/main/java/xdm/app/service/AppConfigService.kt
package xdm.app.service

import xdm.app.ui.components.SortKey

interface AppConfigService {
    fun load()

    fun shouldShowDownloadCompleteWindow(): Boolean

    fun shouldRunVirusScan(): Boolean

    fun shouldRunCommand(): Boolean

    fun shouldShowDownloadProgressWindow(): Boolean

    val defaultDownloadFolder: String?

    val tempFolder: String?

    fun shouldAutoRenameOnConflict(): Boolean

    fun shouldShutdownAfterAllDone(): Boolean

    val maxParallelDownloads: Int
    val recentFolders: List<String?>?
    var autoSelectFolder: Boolean
    val folderIndex: Int
    var sortKey: SortKey?
    var sortAscending: Boolean
    var minVideoSize: Long
}
